pkg/resources/statefulsets/statefulset.go

```diff
@@ -276,56 +279,148 @@ func createResources(r *corev1.ResourceRequirements) corev1.ResourceRequirements
+// complies to the Kubernetes probes requirements.
+func createReadinessProbe(w *wildflyv1alpha1.WildFlyServer) *corev1.Probe {
+	readinessProbeScript, defined := os.LookupEnv("SERVER_READINESS_SCRIPT")
+
+	var initialDelaySeconds int32 = 10
+	var timeoutSeconds int32
+	var periodSeconds int32
+	var successThreshold int32
+	var failureThreshold int32
+
+	if w.Spec.ReadinessProbe != nil {
+		initialDelaySeconds = w.Spec.ReadinessProbe.InitialDelaySeconds
+		timeoutSeconds = w.Spec.ReadinessProbe.TimeoutSeconds
+		periodSeconds = w.Spec.ReadinessProbe.PeriodSeconds
+		successThreshold = w.Spec.ReadinessProbe.SuccessThreshold
+		failureThreshold = w.Spec.ReadinessProbe.FailureThreshold
+	}
+
 	if defined && !w.Spec.BootableJar {
+		readinessProbeScriptFallback, definedFallback := os.LookupEnv("SERVER_READINESS_SCRIPT_FALLBACK")
+		if !definedFallback {
+			readinessProbeScriptFallback = "curl 127.0.0.1:" + strconv.Itoa(int(resources.HTTPManagementPort)) + "/health/ready"
+		}
+		probeScript := "if [ -f " + readinessProbeScript + " ]; then " + readinessProbeScript + "; else " + readinessProbeScriptFallback + "; fi"
 		return &corev1.Probe{
 			Handler: corev1.Handler{
 				Exec: &corev1.ExecAction{
-					Command: []string{"/bin/bash", "-c", livenessProbeScript},
+					Command: []string{"/bin/bash", "-c", probeScript},
 				},
 			},
-			InitialDelaySeconds: 60,
+			InitialDelaySeconds: initialDelaySeconds,
+			TimeoutSeconds:      timeoutSeconds,
+			PeriodSeconds:       periodSeconds,
+			SuccessThreshold:    successThreshold,
+			FailureThreshold:    failureThreshold,
 		}
 	}
 	return &corev1.Probe{
 		Handler: corev1.Handler{
 			HTTPGet: &corev1.HTTPGetAction{
-				Path: "/health/live",
+				Path: "/health/ready",
 				Port: intstr.FromString("admin"),
 			},
 		},
-		InitialDelaySeconds: 60,
+		InitialDelaySeconds: initialDelaySeconds,
+		TimeoutSeconds:      timeoutSeconds,
+		PeriodSeconds:       periodSeconds,
+		SuccessThreshold:    successThreshold,
+		FailureThreshold:    failureThreshold,
 	}
 }
 
-// createReadinessProbe create an Exec probe if the SERVER_READINESS_SCRIPT env var is present
+// createStartupProbe create an Exec probe if the SERVER_LIVENESS_SCRIPT env var is present
 // *and* the application is not using Bootable Jar.
+// Otherwise, it creates a HTTPGet probe that checks the /health/live endpoint on the admin port.
 //
-// If defined, the SERVER_READINESS_SCRIPT env var must be the path of a shell script that
+// If defined, the SERVER_LIVENESS_SCRIPT env var must be the path of a shell script that
 // complies to the Kubernetes probes requirements.
-func createReadinessProbe(w *wildflyv1alpha1.WildFlyServer) *corev1.Probe {
-	readinessProbeScript, defined := os.LookupEnv("SERVER_READINESS_SCRIPT")
-	if defined && !w.Spec.BootableJar {
+// If SERVER_LIVENESS_SCRIPT script does not exist, then the Probe will execute the script defined by SERVER_LIVENESS_SCRIPT_FALLBACK
```

Comment:
the same type of comment should also be there for createReadinessProbe

Reply:
Fixed.